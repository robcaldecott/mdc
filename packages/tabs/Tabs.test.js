import React from "react";
import { TabBar, Tab, TabIcon, TabIconText } from ".";
import renderer from "react-test-renderer";

it("renders text tabs", () => {
  expect(
    renderer.create(
      <TabBar className="custom">
        <Tab active>Tab 1</Tab>
        <Tab>Tab 2</Tab>
        <Tab>Tab 3</Tab>
      </TabBar>
    )
  ).toMatchSnapshot();
});

it("renders icon tabs", () => {
  expect(
    renderer.create(
      <TabBar icons indicator="primary">
        <Tab active>
          <TabIcon>star</TabIcon>
        </Tab>
        <Tab>
          <TabIcon>favorite</TabIcon>
        </Tab>
        <Tab>
          <TabIcon>bookmark</TabIcon>
        </Tab>
      </TabBar>
    )
  ).toMatchSnapshot();
});

it("renders icon and text tabs", () => {
  expect(
    renderer.create(
      <TabBar iconsWithText indicator="accent">
        <Tab active iconAndText>
          <TabIcon>star</TabIcon>
          <TabIconText>Tab 1</TabIconText>
        </Tab>
        <Tab iconAndText>
          <TabIcon>favorite</TabIcon>
          <TabIconText>Tab 2</TabIconText>
        </Tab>
        <Tab iconAndText>
          <TabIcon>bookmark</TabIcon>
          <TabIconText>Tab 3</TabIconText>
        </Tab>
      </TabBar>
    )
  ).toMatchSnapshot();
});
