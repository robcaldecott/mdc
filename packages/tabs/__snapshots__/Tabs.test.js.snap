// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders icon and text tabs 1`] = `
<nav
  className="mdc-tab-bar mdc-tab-bar--icons-with-text"
  indicator="accent"
>
  <a
    className="mdc-tab mdc-tab--active mdc-tab--with-icon-and-text"
  >
    <i
      className="material-icons mdc-tab__icon"
    >
      star
    </i>
    <span
      className="mdc-tab__icon-text"
    >
      Tab 1
    </span>
    <span
      className="mdc-tab__indicator"
    />
  </a>
  <a
    className="mdc-tab mdc-tab--with-icon-and-text"
  >
    <i
      className="material-icons mdc-tab__icon"
    >
      favorite
    </i>
    <span
      className="mdc-tab__icon-text"
    >
      Tab 2
    </span>
    <span
      className="mdc-tab__indicator"
    />
  </a>
  <a
    className="mdc-tab mdc-tab--with-icon-and-text"
  >
    <i
      className="material-icons mdc-tab__icon"
    >
      bookmark
    </i>
    <span
      className="mdc-tab__icon-text"
    >
      Tab 3
    </span>
    <span
      className="mdc-tab__indicator"
    />
  </a>
</nav>
`;

exports[`renders icon tabs 1`] = `
<nav
  className="mdc-tab-bar mdc-tab-bar--icon-tab-bar"
  indicator="primary"
>
  <a
    className="mdc-tab mdc-tab--active"
  >
    <i
      className="material-icons mdc-tab__icon"
    >
      star
    </i>
    <span
      className="mdc-tab__indicator"
    />
  </a>
  <a
    className="mdc-tab"
  >
    <i
      className="material-icons mdc-tab__icon"
    >
      favorite
    </i>
    <span
      className="mdc-tab__indicator"
    />
  </a>
  <a
    className="mdc-tab"
  >
    <i
      className="material-icons mdc-tab__icon"
    >
      bookmark
    </i>
    <span
      className="mdc-tab__indicator"
    />
  </a>
</nav>
`;

exports[`renders text tabs 1`] = `
<nav
  className="mdc-tab-bar custom"
>
  <a
    className="mdc-tab mdc-tab--active"
  >
    Tab 1
    <span
      className="mdc-tab__indicator"
    />
  </a>
  <a
    className="mdc-tab"
  >
    Tab 2
    <span
      className="mdc-tab__indicator"
    />
  </a>
  <a
    className="mdc-tab"
  >
    Tab 3
    <span
      className="mdc-tab__indicator"
    />
  </a>
</nav>
`;
